hexgame/tests/test_board.py

```diff
@@ -0,0 +1,48 @@
+from hexgame.src import board as Board
+from hexgame.src import cell as Cell
+import pytest
+
+
+class TestBoardProperties:
+
+    def test_create_empty_board(self):
+        empty_board = Board.Board(0, 0)
+        assert len(empty_board._board) == 0
+
+    def test_create_small_board(self):
+        small_board = Board.Board(2, 3)
+        assert len(small_board._board) == 2
+        assert len(small_board._board[0]) == 3
+        assert small_board.has_cell(Cell(0, 0))
+        assert small_board.has_cell(Cell(1, 0))
+        assert small_board.has_cell(Cell(0, 1))
+        assert small_board.has_cell(Cell(1, 1))
+        assert small_board.has_cell(Cell(0, 2))
+        assert small_board.has_cell(Cell(1, 2))
+
+    def test_neighbours_of_cell(self):
+        small_board = Board.Board(2, 3)
+        from_bottom_left_corner = small_board.find_neighbours((0, 0))
+        assert len(from_bottom_left_corner) == 3
+        assert Cell(0, 1) in from_bottom_left_corner
+        assert Cell(1, 0) in from_bottom_left_corner
+        assert Cell(1, 1) in from_bottom_left_corner
+        from_bottom_right_corner = small_board.find_neighbours((1, 0))
+        assert len(from_bottom_right_corner) == 2
+        assert Cell(0, 0) in from_bottom_right_corner
+        assert Cell(1, 1) in from_bottom_right_corner
+        from_left_edge = small_board.find_neighbours((0, 1))
+        assert len(from_left_edge) == 4
```

Comment:
ok just one more thing:
now that we have pytest I would acutally rewrite this test by parametrising like this:



@pytest.mark.parametrize("test_input,expected", [({(0,0)}, {}), ({}, {}), ({}, {})])
def test_neighbours_of_cell(test_input, expected):
    assert small_board.find_neighbours(input) =  expected
(i did not write all the thing, but see this [https://docs.pytest.org/en/7.1.x/how-to/parametrize.html](link)